rclpy/src/rclpy/action_server.cpp

```diff
@@ -139,6 +139,13 @@ ActionServer::take_result_request(py::object pymsg_type)
   auto ros_response = convert_from_py(pyresponse); \
   rcl_ret_t ret = rcl_action_send_ ## Type ## _response( \
     rcl_action_server_.get(), header, ros_response.get()); \
+  if (RCL_RET_TIMEOUT == ret) { \
```

Comment:
if timeout happens, we will go ahead to generate the exception in `rclpy`, the exception is caught on the caller side. this just adds the warning message for the user that timeout did happen here.

Reply:
Maybe it is better to move `if (RCL_RET_TIMEOUT == ret)` in `if (RCL_RET_OK != ret)`.  Otherwise, if ret is RCL_RET_OK, the code has to perform the check twice.